estimagic/optimization/optimize.py

```diff
@@ -95,19 +92,99 @@ def minimize(
     criterion,
     params,
     algorithm,
-    criterion_args=None,
     criterion_kwargs=None,
     constraints=None,
     general_options=None,
     algo_options=None,
     dashboard=False,
     db_options=None,
 ):
-    """Minimize *criterion* using *algorithm* subject to *pc* and bounds.
+    """Minimize *criterion* using *algorithm* subject to *constraints* and bounds.
+    Run several optimizations if called by lists of inputs.
+
+    Args:
+        criterion (function or list of functions):
+            Python function that takes a pandas Series with parameters as the first
+            argument and returns a scalar floating point value.
+
+        params (pd.DataFrame or list of pd.DataFrames):
+            See :ref:`params`.
+
+        algorithm (str or list of strings):
+            specifies the optimization algorithm. See :ref:`list_of_algorithms`.
+
+        criterion_kwargs (dict or list of dicts):
+            additional keyword arguments for criterion
+
+        constraints (list or list of lists):
+            list with constraint dictionaries. See for details.
+
+        general_options (dict):
+            additional configurations for the optimization
+
+        algo_options (dict or list of dicts):
+            algorithm specific configurations for the optimization
+
+        dashboard (bool):
+            whether to create and show a dashboard
+
+        db_options (dict):
+            dictionary with kwargs to be supplied to the run_server function.
+
+    """
+
+    arguments, n_opts = process_optimization_arguments(
+        criterion=criterion,
+        params=params,
+        algorithm=algorithm,
+        criterion_kwargs=criterion_kwargs,
+        constraints=constraints,
+        general_options=general_options,
+        algo_options=algo_options,
+        dashboard=dashboard,
+        db_options=db_options,
+    )
+    if n_opts == 1:
+        result = _single_minimize(**arguments)
+    else:
+        if dashboard:
+            raise NotImplementedError(
+                "Dashboard cannot be used for multiple optimizations, yet."
+            )
+
+        # set up multiprocessing
+        if "n_cores" not in arguments["general_options"][0]:
+            raise ValueError(
+                "n_cores need to be specified in general_options"
+                + " if multiple optimizations should be run."
+            )
+        n_cores = arguments["general_options"][0]["n_cores"]
+        pool = Pool(processes=n_cores)
+
+        # `Transpose' arguments and run all optimizations in parallel
+        args_transposed = [{a: arguments[a][i] for a in arguments} for i in range(2)]
```

Comment:
Two questions:
1. Why `range(2)` and not something that depends on the length of the lists in arguments
2. Is there a reason why we can't move the transpose step into `process_arguments`. If not, I would prefer that.